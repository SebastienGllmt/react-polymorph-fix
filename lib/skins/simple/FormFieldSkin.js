'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormFieldSkin = undefined;

var _omit2 = require('lodash/omit');

var _omit3 = _interopRequireDefault(_omit2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FormFieldSkin = exports.FormFieldSkin = function FormFieldSkin(props) {
  return _react2.default.createElement(
    'div',
    {
      className: (0, _classnames2.default)([props.className, props.theme[props.themeId].root, props.disabled ? props.theme[props.themeId].disabled : null, props.error ? props.theme[props.themeId].errored : null])
    },
    props.error && _react2.default.createElement(
      'div',
      { className: props.theme[props.themeId].error },
      props.error
    ),
    props.label && _react2.default.createElement(
      'label',
      {
        role: 'presentation',
        'aria-hidden': true,
        className: props.theme[props.themeId].label,
        onClick: props.focusChild
      },
      props.label
    ),
    _react2.default.createElement(
      'div',
      { className: props.theme[props.themeId].inputWrapper },
      props.render((0, _omit3.default)(props, ['themeId']))
    )
  );
};
//# sourceMappingURL=FormFieldSkin.js.map