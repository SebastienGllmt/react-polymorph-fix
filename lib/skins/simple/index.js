'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map