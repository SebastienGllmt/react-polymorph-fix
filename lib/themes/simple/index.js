'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map