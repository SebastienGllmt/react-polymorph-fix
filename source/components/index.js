// @flow
export const IDENTIFIERS = {
  AUTOCOMPLETE: 'autocomplete',
  BUBBLE: 'bubble',
  BUTTON: 'button',
  CHECKBOX: 'checkbox',
  FLEX: 'flex',
  FORM_FIELD: 'formfield',
  GRID: 'grid',
  GUTTER: 'gutter',
  HEADER: 'header',
  INFINITE_SCROLL: 'infinitescroll',
  INPUT: 'input',
  LOADING_SPINNER: 'loadingspinner',
  MODAL: 'modal',
  OPTIONS: 'options',
  PROGRESS_BAR: 'progressbar',
  RADIO: 'radio',
  SELECT: 'select',
  SWITCH: 'switch',
  TEXT_AREA: 'textarea',
  TOGGLER: 'toggler',
  TOOLTIP: 'tooltip'
};
