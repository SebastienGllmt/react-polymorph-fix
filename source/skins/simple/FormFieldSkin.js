// @flow
import React from 'react';
import type { Element } from 'react';
import { omit } from 'lodash';
import classnames from 'classnames';

type Props = {
  className: string,
  disabled: boolean,
  error: string | Element<any>,
  focusChild: Function,
  label: string | Element<any>,
  onChange: Function,
  render: Function,
  setError: Function,
  theme: Object,
  themeId: string
};

export const FormFieldSkin = (props: Props) => (
  <div
    className={classnames([
      props.className,
      props.theme[props.themeId].root,
      props.disabled ? props.theme[props.themeId].disabled : null,
      props.error ? props.theme[props.themeId].errored : null
    ])}
  >
    {props.error && (
      <div className={props.theme[props.themeId].error}>{props.error}</div>
    )}
    {props.label && (
      <label
        role="presentation"
        aria-hidden
        className={props.theme[props.themeId].label}
        onClick={props.focusChild}
      >
        {props.label}
      </label>
    )}
    <div className={props.theme[props.themeId].inputWrapper}>
      {props.render(omit(props, ['themeId']))}
    </div>
  </div>
);
