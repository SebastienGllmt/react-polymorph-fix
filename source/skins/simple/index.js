// @flow
import { IDENTIFIERS } from '../../components';
import { AutocompleteSkin } from './AutocompleteSkin';
import { BubbleSkin } from './BubbleSkin';
import { ButtonSkin } from './ButtonSkin';
import { CheckboxSkin } from './CheckboxSkin';
import { FormFieldSkin } from './FormFieldSkin';
import { HeaderSkin } from './HeaderSkin';
import { InfiniteScrollSkin } from './InfiniteScrollSkin';
import { InputSkin } from './InputSkin';
import { LoadingSpinnerSkin } from './LoadingSpinnerSkin';
import { ModalSkin } from './ModalSkin';
import { OptionsSkin } from './OptionsSkin';
import { ProgressBarSkin } from './ProgressBarSkin';
import { RadioSkin } from './RadioSkin';
import { SelectSkin } from './SelectSkin';
import { SwitchSkin } from './SwitchSkin';
import { TextAreaSkin } from './TextAreaSkin';
import { TogglerSkin } from './TogglerSkin';
import { TooltipSkin } from './TooltipSkin';

export const SimpleSkins = {
  [IDENTIFIERS.AUTOCOMPLETE]: AutocompleteSkin,
  [IDENTIFIERS.BUBBLE]: BubbleSkin,
  [IDENTIFIERS.BUTTON]: ButtonSkin,
  [IDENTIFIERS.CHECKBOX]: CheckboxSkin,
  [IDENTIFIERS.FORM_FIELD]: FormFieldSkin,
  [IDENTIFIERS.HEADER]: HeaderSkin,
  [IDENTIFIERS.INFINITE_SCROLL]: InfiniteScrollSkin,
  [IDENTIFIERS.INPUT]: InputSkin,
  [IDENTIFIERS.LOADING_SPINNER]: LoadingSpinnerSkin,
  [IDENTIFIERS.MODAL]: ModalSkin,
  [IDENTIFIERS.OPTIONS]: OptionsSkin,
  [IDENTIFIERS.PROGRESS_BAR]: ProgressBarSkin,
  [IDENTIFIERS.RADIO]: RadioSkin,
  [IDENTIFIERS.SELECT]: SelectSkin,
  [IDENTIFIERS.SWITCH]: SwitchSkin,
  [IDENTIFIERS.TEXT_AREA]: TextAreaSkin,
  [IDENTIFIERS.TOGGLER]: TogglerSkin,
  [IDENTIFIERS.TOOLTIP]: TooltipSkin
};
