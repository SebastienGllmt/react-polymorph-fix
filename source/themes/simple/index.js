// @flow
// import theme IDENTIFIERS constants
import { IDENTIFIERS } from '../../components';

// css modules plugin converts all imports below into plain objects
import SimpleAutocomplete from './SimpleAutocomplete.scss';
import SimpleBubble from './SimpleBubble.scss';
import SimpleButton from './SimpleButton.scss';
import SimpleCheckbox from './SimpleCheckbox.scss';
import SimpleFlex from './SimpleFlex.scss';
import SimpleFormField from './SimpleFormField.scss';
import SimpleGrid from './SimpleGrid.scss';
import SimpleGutter from './SimpleGutter.scss';
import SimpleHeader from './SimpleHeader.scss';
import SimpleInfiniteScroll from './SimpleInfiniteScroll.scss';
import SimpleInput from './SimpleInput.scss';
import SimpleLoadingSpinner from './SimpleLoadingSpinner.scss';
import SimpleModal from './SimpleModal.scss';
import SimpleOptions from './SimpleOptions.scss';
import SimpleProgressBar from './SimpleProgressBar.scss';
import SimpleRadio from './SimpleRadio.scss';
import SimpleSelect from './SimpleSelect.scss';
import SimpleSwitch from './SimpleSwitch.scss';
import SimpleTextArea from './SimpleTextArea.scss';
import SimpleToggler from './SimpleToggler.scss';
import SimpleTooltip from './SimpleTooltip.scss';

// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
export const SimpleTheme = {
  [IDENTIFIERS.AUTOCOMPLETE]: SimpleAutocomplete,
  [IDENTIFIERS.BUBBLE]: SimpleBubble,
  [IDENTIFIERS.BUTTON]: SimpleButton,
  [IDENTIFIERS.CHECKBOX]: SimpleCheckbox,
  [IDENTIFIERS.FLEX]: SimpleFlex,
  [IDENTIFIERS.FORM_FIELD]: SimpleFormField,
  [IDENTIFIERS.GRID]: SimpleGrid,
  [IDENTIFIERS.GUTTER]: SimpleGutter,
  [IDENTIFIERS.HEADER]: SimpleHeader,
  [IDENTIFIERS.INFINITE_SCROLL]: SimpleInfiniteScroll,
  [IDENTIFIERS.INPUT]: SimpleInput,
  [IDENTIFIERS.LOADING_SPINNER]: SimpleLoadingSpinner,
  [IDENTIFIERS.MODAL]: SimpleModal,
  [IDENTIFIERS.OPTIONS]: SimpleOptions,
  [IDENTIFIERS.PROGRESS_BAR]: SimpleProgressBar,
  [IDENTIFIERS.RADIO]: SimpleRadio,
  [IDENTIFIERS.SELECT]: SimpleSelect,
  [IDENTIFIERS.SWITCH]: SimpleSwitch,
  [IDENTIFIERS.TEXT_AREA]: SimpleTextArea,
  [IDENTIFIERS.TOGGLER]: SimpleToggler,
  [IDENTIFIERS.TOOLTIP]: SimpleTooltip
};
