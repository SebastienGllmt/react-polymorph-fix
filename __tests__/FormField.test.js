import React from 'react';

import { FormField } from '../source/components/FormField';
import { FormFieldSkin } from '../source/skins/simple/FormFieldSkin';
import { renderInSimpleTheme } from './helpers/theming';

const renderFormField = () => <div className="render-prop" />;

test('FormField renders correctly', () => {
  const component = renderInSimpleTheme(
    <FormField
      skin={FormFieldSkin}
      render={renderFormField}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FormField renders with label', () => {
  const component = renderInSimpleTheme(
    <FormField
      label="Add a Label"
      skin={FormFieldSkin}
      render={renderFormField}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FormField renders with an error', () => {
  const component = renderInSimpleTheme(
    <FormField
      error="Add an Error"
      skin={FormFieldSkin}
      render={renderFormField}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FormField is disabled', () => {
  const component = renderInSimpleTheme(
    <FormField
      disabled
      skin={FormFieldSkin}
      render={({ disabled }) => <span>{disabled.toString()}</span>}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FormField renders an input element', () => {
  const component = renderInSimpleTheme(
    <FormField
      skin={FormFieldSkin}
      render={() => <input className="render-prop" />}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
