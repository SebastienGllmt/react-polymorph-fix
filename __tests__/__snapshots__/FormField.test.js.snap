// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField is disabled 1`] = `
<div
  className="root disabled"
>
  
  <div
    className="inputWrapper"
  >
    <span>
      true
    </span>
  </div>
</div>
`;

exports[`FormField renders an input element 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="render-prop"
    />
  </div>
</div>
`;

exports[`FormField renders correctly 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <div
      className="render-prop"
    />
  </div>
</div>
`;

exports[`FormField renders with an error 1`] = `
<div
  className="root errored"
>
  <div
    className="error"
  >
    Add an Error
  </div>
  <div
    className="inputWrapper"
  >
    <div
      className="render-prop"
    />
  </div>
</div>
`;

exports[`FormField renders with label 1`] = `
<div
  className="root"
>
  
  <label
    aria-hidden={true}
    className="label"
    onClick={[Function]}
    role="presentation"
  >
    Add a Label
  </label>
  <div
    className="inputWrapper"
  >
    <div
      className="render-prop"
    />
  </div>
</div>
`;
